testsupport/wait/host.go

```diff
@@ -704,6 +713,99 @@ func containsUserAccountStatus(uaStatuses []toolchainv1alpha1.UserAccountStatusE
+		err = a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj)
+		if err != nil && !errors.IsNotFound(err) {
+			// return the error
+			return false, err
+		} else if errors.IsNotFound(err) {
+			// keep waiting
+			return false, nil
+		}
+		tier = obj
+		return matchUserTierWaitCriterion(obj, criteria...), nil
+	})
+	// no match found, print the diffs
+	if err != nil {
+		a.printUserTierWaitCriterionDiffs(tier, criteria...)
+	}
+	return tier, err
+}
+
+// UserTierWaitCriterion a struct to compare with an expected UserTier
+type UserTierWaitCriterion struct {
+	Match func(*toolchainv1alpha1.UserTier) bool
+	Diff  func(*toolchainv1alpha1.UserTier) string
+}
+
+func matchUserTierWaitCriterion(actual *toolchainv1alpha1.UserTier, criteria ...UserTierWaitCriterion) bool {
+	for _, c := range criteria {
+		// if at least one criteria does not match, keep waiting
+		if !c.Match(actual) {
+			return false
+		}
+	}
+	return true
+}
+
+func (a *HostAwaitility) printUserTierWaitCriterionDiffs(actual *toolchainv1alpha1.UserTier, criteria ...UserTierWaitCriterion) {
+	buf := &strings.Builder{}
+	if actual == nil {
+		buf.WriteString("failed to find UserTier\n")
+	} else {
+		buf.WriteString("failed to find UserTier with matching criteria:\n")
+		buf.WriteString("actual:\n")
+		y, _ := StringifyObject(actual)
+		buf.Write(y)
+		buf.WriteString("\n----\n")
+		buf.WriteString("diffs:\n")
+		for _, c := range criteria {
+			if !c.Match(actual) {
+				buf.WriteString(c.Diff(actual))
+				buf.WriteString("\n")
+			}
+		}
+	}
+	// also include other resources relevant in the host namespace, to help troubleshooting
+	buf.WriteString(a.sprintAllResources())
+
+	a.T.Log(buf.String())
+}
+
+// UntilUserTierHasDeactivationTimeoutDays verify that the UserTier status.Updates has the specified number of entries
+func UntilUserTierHasDeactivationTimeoutDays(expected int) UserTierWaitCriterion {
+	return UserTierWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.UserTier) bool {
+			return actual.Spec.DeactivationTimeoutDays == expected
+		},
+		Diff: func(actual *toolchainv1alpha1.UserTier) string {
+			return fmt.Sprintf("expected deactivationTimeoutDay value %d. Actual: %d", expected, actual.Spec.DeactivationTimeoutDays)
+		},
+	}
+}
+
+// UntilUserTierHasGeneration verify that the UserTier status.Updates has the specified number of entries
+func UntilUserTierHasGeneration(expected int64) UserTierWaitCriterion {
+	return UserTierWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.UserTier) bool {
+			return actual.Generation == expected
+		},
+		Diff: func(actual *toolchainv1alpha1.UserTier) string {
+			return fmt.Sprintf("expected generation value %d. Actual: %d", expected, actual.Generation)
+		},
+	}
+}
+
+func (a *HostAwaitility) WaitUntilBaseUserTierIsUpdated() error {
+	_, err := a.WaitForUserTier("deactivate30", UntilUserTierHasDeactivationTimeoutDays(30), UntilUserTierHasGeneration(2))
```

Comment:
Is checking the `Generation` number necessary? isn't it enough to check that the deactivation value has been updated? The only thing that we care about is that the deactivation value is set to the expected one, we don't care how many times the resource was updated.
```suggestion

func (a *HostAwaitility) WaitUntilBaseUserTierIsUpdated() error {
	_, err := a.WaitForUserTier("deactivate30", UntilUserTierHasDeactivationTimeoutDays(30))
```

Reply:
I see, then maybe change the check so it's not equal to `1`. But it's not super important - I guess that we don't do that in NSTemplateTier either.